parser/openfoodfacts_taxonomy_parser/parser.py

```diff
@@ -0,0 +1,283 @@
+                line = line.replace("’", "'")
+                # replace commas that have no space around by a lower comma character 
+                # and do the same for escaped comma (preceded by a \) 
+                # (to distinguish them from commas acting as tags separators)
+                line = re.sub(r"(\d),(\d)", r"\1‚\2", line)
+                line = re.sub(r"\\,", "\\‚", line)
+                #removes parenthesis for roman numeral
+                line = re.sub(r"\(([ivx]+)\)", r"\1", line, flags=re.I)
+                yield line_number,line
+        yield line_number,"" #to end the last entry if not ended
+
+    def normalizing(self,line,lang="default"):
+        """normalize a string depending of the language code lang"""
+        line = unicodedata.normalize("NFC", line)
+        if lang in ['fr','ca','es','it','nl','pt','sk','en']: #removing accent
+            line = re.sub(r"[¢£¤¥§©ª®°²³µ¶¹º¼½¾×‰€™]", "-", line)
+            line = unidecode.unidecode(line)
+        if lang not in [ ]: #lower case except if language in list
+            line=line.lower()
+        line = re.sub(r"[\u0000-\u0027\u200b]", "-", line)
+        line = re.sub(r"&\w+;", "-", line)
+        line = re.sub(r"[\s!\"#\$%&'()*+,\/:;<=>?@\[\\\]^_`{\|}~¡¢£¤¥¦§¨©ª«¬®¯°±²³´µ¶·¸¹º»¼½¾¿×ˆ˜–—‘’‚“”„†‡•…‰‹›€™\t]", "-", line)
+        line = re.sub(r"-+", "-", line)
+        line = line.strip("-")
+        return line
+
+    def remove_stopwords(self,lc,words):
+        """to remove the stopwords that were read at the beginning of the file"""
+        if lc in self.stopwords :
+            words_to_remove = self.stopwords[lc]
+            new_words = []
+            for word in words.split("-"):
+                if word not in words_to_remove :
+                    new_words.append(word)
+            return ("-").join(new_words)
+        else : return words
+
+    def add_line(self,line):
+        """to get a normalized string but keeping the language code "lc:" and the "," (commas) separators , used to add an id to entries and to add a parent tag"""
+        if len(line)<=3:return self.normalizing(line)
+        else:
+            lang=line[:2]
+            new_line=lang+":"
+            for word in line[3:].split(","):
+                new_line+=self.remove_stopwords( lang,self.normalizing(word,lang) ) +','
+            new_line=new_line[:-1]
+            return new_line
+
+    def get_lc_value(self,line):
+        """to get the language code "lc" and a list of normalized values"""
+        lc=line[:2]
+        new_line=[]
+        for word in line[3:].split(","):
+            new_line.append( self.remove_stopwords( lc,self.normalizing(word,lc) ) )
+        return lc,new_line
+
+    def new_node_data(self):
+        """ To create an empty dictionary that will be used to create node """
+        data = {
+                "id" : '',
+                "main_language" : '',
+                "comment" : [],
```

Comment:
@BryanH01 @alexgarel Isn't this property is called "preceding_lines"? If so, please do change it everywhere.
```suggestion
                "preceding_lines" : [],
```

Reply:
yes that's `preceding_lines`.